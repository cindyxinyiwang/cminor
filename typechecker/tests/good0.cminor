i : integer = 1;
j : integer = i;

func: function integer (i: integer, j: integer) = {
	i = i+1;
	j = i+1;

	x: integer = 1;
	y: boolean;
	y = true;
	j = x;
	{
		i: integer = j;
		i > 0;
	}
}

myFunc: function void (m: boolean, n: integer) = {
	i = n +1;

	n = func(1, 2);
	myFunc(true, 2);
	func(i, j);
}
