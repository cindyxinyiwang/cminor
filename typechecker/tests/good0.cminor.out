i resovles to global i
i resovles to param 0
i resovles to param 0
j resovles to param 1
i resovles to param 0
y resovles to local 1
j resovles to param 1
x resovles to local 0
j resovles to param 1
i resovles to local 2
i resovles to global i
n resovles to param 1
n resovles to param 1
func resovles to global func
myFunc resovles to global myFunc
i resovles to global i
j resovles to global j
func resovles to global func
